projects/102_Test/src/main.c

```diff
@@ -0,0 +1,42 @@
+#include <stdint.h>
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void prv_counter_timeout(SoftTimerId timer_id, void *context) {
+  Counters *counter = context;
+
+  counter->counter_a++;
+  LOG_DEBUG("Counter A: %u\n", (counter->counter_a));
+
+  if ((counter->counter_a) % 2 == 0) {
+    counter->counter_b++;
+    LOG_DEBUG("Counter B: %u\n", (counter->counter_b));
+  }
+
+  soft_timer_start_seconds(0.5, prv_counter_timeout, counter, NULL);
```

Comment:
I'm kind of surprised this works, a more conventional way to do it would be `soft_timer_start_millis(500, prv_counter_timeout, counter, NULL)` but this works too. In any case, please factor out the 0.5 constant that you use twice into a `#define`'d constant with an appropriate name that specifies the unit (seconds) - this makes it easier to read and change.